server/constructors.go

```diff
@@ -27,7 +28,7 @@ type (
 
 	// AppCreator is a function that allows us to lazily initialize an
 	// application using various configurations.
-	AppCreator func(log.Logger, dbm.DB, io.Writer) Application
+	AppCreator func(log.Logger, dbm.DB, io.Writer, *viper.Viper) Application
```

Comment:
Is it possible to remove this viper reference in the AppCreator? Something inside me feels this is not good <3

Reply:
Done. Replaced with `AppOptions`.